src/components/raycaster.js

```diff
@@ -60,25 +63,54 @@ module.exports.Component = registerComponent('raycaster', {
   },
 
   /**
+   * Update list of objects to test for intersection once child is loaded.
+   */
+  refreshOnceChildLoaded: function (evt) {
+    var self = this;
+    var el = evt.detail.el;
+    if (!el) { return; }
+    if (el.hasLoaded) { this.refreshObjects(); } else {
+      el.addEventListener('loaded', function nowRefresh (evt) {
+        el.removeEventListener('loaded', nowRefresh);
+        self.refreshObjects();
+      });
+    }
+  },
+
+  /**
    * Update list of objects to test for intersection.
    */
   refreshObjects: function () {
+    var children;
     var data = this.data;
     var i;
-    var objectEls;
+    var j;
+    var object;
+    var objects;
+    var objectsAreEls = !!data.objects;
 
     // Push meshes onto list of objects to intersect.
-    if (data.objects) {
-      objectEls = this.el.sceneEl.querySelectorAll(data.objects);
-      this.objects = [];
-      for (i = 0; i < objectEls.length; i++) {
-        this.objects.push(objectEls[i].object3D);
-      }
-      return;
+    // If objects not defined, intersect with everything.
+    if (objectsAreEls) {
+      objects = this.el.sceneEl.querySelectorAll(data.objects);
+    } else {
+      objects = this.el.sceneEl.object3D.children;
     }
 
-    // If objects not defined, intersect with everything.
-    this.objects = this.el.sceneEl.object3D.children;
+    this.objects = [];
+    for (i = 0; i < objects.length; i++) {
+      // If we were given elements, get the object3D.
+      object = objectsAreEls ? objects[i].object3D : objects[i];
+      // A-Frame wraps everything (e.g. in a Group) so we want children.
+      children = object.children;
+      if (children && children.length) {
+        // Add the object3D's children so non-recursive raycasting will work correctly.
+        for (j = 0; j < children.length; j++) { this.objects.push(children[j]); }
+      } else {
+        // If there aren't any children, then until a refresh after geometry loads,
+        // raycast won't see this object... but that should happen automatically.
+      }
+    }
```

Comment:
Heh, ok it's more complicated than I thought, there are "live" and "static" NodeLists, and the result of querySelectorAll() is the latter, which means it can probably be hoisted anyway. Live NodeLists are more trouble. Feel free to ignore my concern and I'll just patch it if I actually run into a noticeable performance issue. :)